### PR TITLE
Refactor BaseFileSystemMaster

### PR DESCRIPTION
**Pull request by elega**
### What changes are proposed in this pull request?

Refactor BaseFileSystemMaster and its test, a test base is extracted out

### Why are the changes needed?

So that the class can be inherited easily

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
N/A

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java [hunk 1]
```diff
@@ -617,22 +621,38 @@ <R> R rpc(RpcCallable<FileSystemMasterClient, R> fn)
...
+    }
+    if (e instanceof UnavailableException || e instanceof UnauthenticatedException) {
       throw e;
-    } catch (AlluxioStatusException e) {
-      throw e.toAlluxioException();
     }
+    throw e.toAlluxioException();
+  }
+
+  /**
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java:651]: nit: If there's anything you prefer sharing context for (or something warnings while sharing a context), let's mention here a bit

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment, thanks!

File: core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java (hunk 1 above)
Comment (elega) [reply core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java:651]: n/a don't think there's anything worth mentioning

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**